app/assemblers/assemble_profile_testimonials_list.rb

```diff
@@ -18,7 +20,22 @@ def call
   end
 
   memoize
-  def testimonials
-    user.mentor_testimonials.published
+  def paginated_testimonials
+    page = testimonials.order(id: :desc).page(params[:page]).per(self.class.testimonials_per_page)
+    return page unless params[:uuid]
+
+    page_uuids = page.map(&:uuid)
+    return page if page_uuids.include?(params[:uuid])
+
+    selected = user.mentor_testimonials.published.find_by(uuid: params[:uuid])
+    return page unless selected
+
+    page_uuids.unshift(selected.uuid) # Show selected testimonial first
+
+    testimonials.where(uuid: page_uuids).
+      order(Arel.sql("FIND_IN_SET(uuid, '#{page_uuids.join(',')}')")).
+      page(page.current_page).per(page.size + 1)
```

Comment:
I've kept the same ordering to make things consistent-like

Reply:
Lovely.